sw/device/silicon_creator/lib/otbn_boot_services.c

```diff
@@ -0,0 +1,190 @@
+OTBN_DECLARE_SYMBOL_ADDR(boot, x);          // ECDSA public key x-coordinate.
+OTBN_DECLARE_SYMBOL_ADDR(boot, y);          // ECDSA public key y-coordinate.
+OTBN_DECLARE_SYMBOL_ADDR(boot, r);          // ECDSA signature component r.
+OTBN_DECLARE_SYMBOL_ADDR(boot, s);          // ECDSA signature component s.
+OTBN_DECLARE_SYMBOL_ADDR(
+    boot, attestation_additional_seed);  // Additional seed for ECDSA keygen.
+
+static const otbn_app_t kOtbnAppBoot = OTBN_APP_T_INIT(boot);
+static const otbn_addr_t kOtbnVarBootMode = OTBN_ADDR_T_INIT(boot, mode);
+static const otbn_addr_t kOtbnVarBootRsaMod = OTBN_ADDR_T_INIT(boot, rsa_mod);
+static const otbn_addr_t kOtbnVarBootRsaM0inv =
+    OTBN_ADDR_T_INIT(boot, rsa_m0inv);
+static const otbn_addr_t kOtbnVarBootRsaInout =
+    OTBN_ADDR_T_INIT(boot, rsa_inout);
+static const otbn_addr_t kOtbnVarBootMsg = OTBN_ADDR_T_INIT(boot, msg);
+static const otbn_addr_t kOtbnVarBootX = OTBN_ADDR_T_INIT(boot, x);
+static const otbn_addr_t kOtbnVarBootY = OTBN_ADDR_T_INIT(boot, y);
+static const otbn_addr_t kOtbnVarBootR = OTBN_ADDR_T_INIT(boot, r);
+static const otbn_addr_t kOtbnVarBootS = OTBN_ADDR_T_INIT(boot, s);
+static const otbn_addr_t kOtbnVarBootAttestationAdditionalSeed =
+    OTBN_ADDR_T_INIT(boot, attestation_additional_seed);
+
+enum {
+  /*
+   * Mode is represented by a single word.
+   */
+  kOtbnBootModeWords = 1,
+  /*
+   * Mode to run RSA modular exponentiation.
+   *
+   * Value taken from `boot.s`.
+   */
+  kOtbnBootModeSecBootModexp = 0x7d3,
+  /*
+   * Mode to generate an attestation keypair.
+   *
+   * Value taken from `boot.s`.
+   */
+  kOtbnBootModeAttestationKeygen = 0x2bf,
+  /*
+   * Mode to endorse a message with a saved private key.
+   *
+   * Value taken from `boot.s`.
+   */
+  kOtbnBootModeAttestationEndorse = 0x5e8,
+  /*
+   * Mode to save an attesation private key.
+   *
+   * Value taken from `boot.s`.
+   */
+  kOtbnBootModeAttestationKeySave = 0x64d,
+};
+
+rom_error_t otbn_boot_app_load(void) { return otbn_load_app(kOtbnAppBoot); }
+
+rom_error_t otbn_boot_attestation_keygen(
+    const attestation_seed_t *additional_seed,
+    keymgr_diversification_t diversification,
+    attestation_public_key_t *public_key) {
+  // Trigger key manager to sideload the attestation key into OTBN.
+  HARDENED_RETURN_IF_ERROR(
+      keymgr_generate_attestation_key_otbn(diversification));
+
+  // Write the mode.
+  uint32_t mode = kOtbnBootModeAttestationKeygen;
+  HARDENED_RETURN_IF_ERROR(
+      otbn_dmem_write(kOtbnBootModeWords, &mode, kOtbnVarBootMode));
+
+  // Write the additional seed.
+  HARDENED_RETURN_IF_ERROR(
+      otbn_dmem_write(kAttestationSeedWords, additional_seed->seed,
+                      kOtbnVarBootAttestationAdditionalSeed));
+
+  // Run the OTBN program (blocks until OTBN is done).
+  HARDENED_RETURN_IF_ERROR(otbn_execute());
+  SEC_MMIO_WRITE_INCREMENT(kOtbnSecMmioExecute);
+
+  // TODO: Check the instruction count register (see `mod_exp_otbn`).
```

Comment:
are these still relevant?

Reply:
Yes, created https://github.com/lowRISC/opentitan/issues/20024 to track them. I'll update the TODO to reference that issue.